argo/providers/vai/provider.go

```diff
@@ -193,8 +202,38 @@ func (vaip VAIProvider) DeletePipeline(ctx context.Context, providerConfig VAIPr
 	return nil
 }
 
+func ExtractFromMap[T any](targetMap map[string]any, fieldName string) (result T, err error) {
```

Comment:
How about calling this `ExtractTypedFromMap`? I wasn't quite sure what this was supposed to do when I first looked at this function.
We could also move this into a separate file as it is not specific to this provider but considering that we are only using it here it is fine for now

Reply:
actually, what is the reason for making this public? imo it should either be public in a separate file or privaye here

Reply:
Mistake at moment is that it's public, will change, as you say if more usage will move it out